crates/networking/p2p/rlpx/connection.rs

```diff
@@ -139,7 +142,48 @@ impl<S: AsyncWrite + AsyncRead + std::marker::Unpin> RLPxConnection<S> {
         ))
     }
 
-    pub async fn handshake(&mut self) -> Result<(), RLPxError> {
+    pub async fn start_peer(&mut self, table: Arc<Mutex<crate::kademlia::KademliaTable>>) {
+        // Perform handshake
+        if let Err(e) = self.handshake().await {
+            self.peer_conn_failed("Handshake failed", e, table).await;
+        } else {
+            // Handshake OK: handle connection
+            if let Err(e) = self.handle_peer_conn().await {
+                self.peer_conn_failed("Error during RLPx connection", e, table)
```

Comment:
Does this block until the connection finishes? If that's the case we should comment it in the start_peer documentation.